client/src/pages/metadata-query/data-source-query.tsx

```diff
@@ -0,0 +1,106 @@
+import { useGetDatasources } from '@/api/datasource/hooks/use-get-datasources';
+import React, { useEffect, useState } from 'react';
+
+export const SourceQuery = ({ validator, queryName, handleQueryValid, handleQueryInvalid }) => {
+  const { apiQuery, blobQuery } = useGetDatasources();
+  const [selections, setSelections] = useState({});
+
+  useEffect(() => {
+    if (apiQuery.data || blobQuery.data) {
+      setDataSources();
+    }
+  }, []);
```

Comment:
Instead of having an empty dependency array woudl make more sense to have a dependency on apiquery.data and blobquery.data

Reply:
I initially had it as that, but this use effect was initialises the data sources to all be checked as false. When I had the data in the dependency array they were set to being unchecked on every re-render, so this empty array makes it so they're only set as unchecked on the first render

Reply:
The problem with this approach it is that you might start with them empty but them the result from the query might come in and what about them now ? You will not update the page to have the actual values.

Reply:
What would you suggest? Having the dependency on apiquery.data breaks it, are there other options?